Code/Engine/RendererDX11/CommandEncoder/Implementation/CommandEncoderImplDX11.cpp

```diff
@@ -129,32 +129,37 @@ void ezGALCommandEncoderImplDX11::SetUnorderedAccessViewPlatform(ezUInt32 uiSlot
 
 // Fence & Query functions
 
-void ezGALCommandEncoderImplDX11::InsertFencePlatform(const ezGALFence* pFence)
+void ezGALCommandEncoderImplDX11::SignalFencePlatform(const ezGALFence* pFence, ezUInt64 value)
 {
-  m_pDXContext->End(static_cast<const ezGALFenceDX11*>(pFence)->GetDXFence());
+  //m_pDXContext->End(static_cast<const ezGALFenceDX11*>(pFence)->GetDXFence());
+  pFence->Signal(value);
 }
 
-bool ezGALCommandEncoderImplDX11::IsFenceReachedPlatform(const ezGALFence* pFence)
+bool ezGALCommandEncoderImplDX11::IsFenceReachedPlatform(const ezGALFence* pFence, ezUInt64 value)
 {
-  BOOL data = FALSE;
-  if (m_pDXContext->GetData(static_cast<const ezGALFenceDX11*>(pFence)->GetDXFence(), &data, sizeof(data), 0) == S_OK)
-  {
-    EZ_ASSERT_DEV(data == TRUE, "Implementation error");
-    return true;
-  }
+  //BOOL data = FALSE;
+  //if (m_pDXContext->GetData(static_cast<const ezGALFenceDX11*>(pFence)->GetDXFence(), &data, sizeof(data), 0) == S_OK)
+  //{
+  //  EZ_ASSERT_DEV(data == TRUE, "Implementation error");
+  //  return true;
+  //}
+
+  //return false;
 
-  return false;
+  return pFence->GetCompletedValue() == value; // >=?(but what if value overflows?), 
```

Comment:
This has to be >= otherwise is won't work if GPU is too fast or CPU too slow. Overflow is not a problem for a 64bit value which is typically increased once per frame.